client/spec/search_spec.js

```diff
@@ -39,6 +39,33 @@ describe('Search', function() {
         expect(element.all(by.repeater('items._items')).count()).toBe(1);
         expect(element.all(by.repeater('parameter in tags.selectedKeywords')).count()).toBe(1);
     });
+    
+    it('can search by subject codes field', function() {
+        workspace.switchToDesk('SPORTS DESK').then(content.setListView);
```

Comment:
In protractor docs is specified that '.then()' is not needed and most (but not all :) ) of the tests works without it.
Probably it is better to avoid them and use only where problems appears. 
@petrjasek Probably we have to decide about the style used to write e2e tests.
Also all selections and actions should be moved on a helper file. This is an issue for all tests from this file, so when/if it will be done, it should be done for all tests.